packages/utils/src/validation/yup.js

```diff
@@ -127,9 +127,52 @@ const testSync = (schema, createError) =>
     },
   });
 
+class CustomLazy extends Lazy {
```

Comment:
Also, can we please break out into a separate file 🙏

Reply:
One last request, could we possibly get a couple of unit tests for this class? I think it would be really helpful to give people a better understanding of what it's for, and will make extending it easier in the future 👍